### PR TITLE
ShareModal: Share externally

### LINKED ISSUE #85268: Share externally option (old public dashboard feature)
**Before**

**- Specific people**
Image

**- Anyone with a link**
Image

**Now**

**- Specific people**
  <video src="https://github.com/grafana/grafana/assets/51436795/5ccab44b-a3ce-48b6-8bbe-97752acf09eb" />

**- Anyone with a link**
<video src="https://github.com/grafana/grafana/assets/51436795/8acbb95a-d475-44c9-bc11-8a5271acc099" />

**We are changing with this task:**

**General access for this feature**
- Having this option in the dropdown to be **"Share externally"**
- Open a drawer with a select menu for **"Only specific people"** or **"Anyone with a link"**

**Specific people option**

- A notification message at the first time you enter about the cost per active user
- Limitations like template variables and data sources, remain as an inline notification at the top
- **Invite field**, by adding their email, it will trigger a confirmation dialog before sending
- People with access `tab`, list of people invited to view this dashboard, with the actions of (**resend invite** or **revoke access**)
- Settings `tab` with the following config as `toggle`:
      - **Set time range** (Include the current time range in the URL)
      - **Enable time range** (Allow people to change time range)
      - **Display annotations** (Present annotations on this Dashboard)
- Actions `buttons` with a lead `icon` are at the bottom of the content
      - **Copy link:** it will copy to your clipboard the link of this dashboard
      - **Revoke access:** revoking the current link, and it won't longer works until you create a new link and make it shareable again.
      - **Pause access:** pausing will temporarily disable access to this dashboard for all users.

**Anyone with the link option**

- A notification inline banner about informing the user that this dashboard is going to be public
- Settings, same as above
- Actions `buttons` with a lead `icon` are at the bottom of the content
      - **Copy public URL:** it will copy to your clipboard the link of this dashboard
      - **Revoke public access:** revoking the current link, and it won't longer works until you create a new public link and make it shareable again.
      - **Pause access:** pausing will temporarily disable access to this dashboard for all users.

**Resources**
[Figma](https://www.figma.com/file/AjvrAorYpnfSoMpU5kwf39/Adoption-of-Sharing?type=design&node-id=3118-42711&mode=design)

### PR DESCRIPTION
**Pull request by juanicabanas**
**What is this feature?**

It adds "Share externally" feature inside new Share Modal redesign

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/85268

**Special notes for your reviewer:**

Enable `newDashboardSharingComponent` FF

[Enterprise PR](https://github.com/grafana/grafana-enterprise/pull/6672)

| Share externally button  | Drawer content |
| ------------- | ------------- |
| image | image  |

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx
Comment (dprokop) [inline public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx:33]: Don't think you need to pass the ref since this is a scene object. You can access the dashboard using scene utils (`getDashboardSceneFor`). The case when passing ref may be beneficial is tests mainly as you don't need to prep the entire structure to get the dashboard scene in place.

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx
Comment (dprokop) [inline public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx:141]: Do you need to pass dashboard to RTK queries? I would expect the RTK queries to perform server comms only, while the side effects are handled externally. Otherwise those queries can't be tested in isolation.

I would expect these kinds of functions to look like this:

```ts
const onDeleteClick = async () => {
    DashboardInteractions.revokePublicDashboardClicked();
    await deletePublicDashboard({ 
        uid: publicDashboard!.uid,
        dashboardUid: dashboard.state.uid!,
    });
   
    dashboard.onNAME_OF THE_SIDE_EFFECT(...)
}
```

**Comment by juanicabanas** [comment]
/deploy-to-hg --enterprise-ref juanicabanas/share-externally-sharing-redesign

**Comment by juanicabanas** [comment]
/deploy-to-hg --enterprise-ref juanicabanas/share-externally-sharing-redesign

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/EmailShare/ConfigEmailSharing/EmailListConfiguration.tsx
Comment (evictorero) [inline public/app/features/dashboard-scene/sharing/ShareButton/share-externally/EmailShare/ConfigEmailSharing/EmailListConfiguration.tsx:31]: Shouldn't these labels be translated?

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/EmailShare/ConfigEmailSharing/EmailListConfiguration.tsx
Comment (evictorero) [inline public/app/features/dashboard-scene/sharing/ShareButton/share-externally/EmailShare/ConfigEmailSharing/EmailListConfiguration.tsx:32]: Shouldn't these labels be translated?

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareTypeSelect.tsx
Comment (evictorero) [inline public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareTypeSelect.tsx:102]: When the first option is selected, the difference in colors is barely noticeable when hovering over options.

File: public/app/features/dashboard/api/publicDashboardApi.ts
Comment (evictorero) [inline public/app/features/dashboard/api/publicDashboardApi.ts:203]: In general, I found several messages without translation. Are we planning to do it later, or should it be part of this PR?

File: public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils.ts
Comment (evictorero) [inline public/app/features/dashboard/components/ShareModal/SharePublicDashboard/SharePublicDashboardUtils.ts:104]: Why is this code duplicated? We have the same in `utils.ts`

**Review by evictorero — COMMENTED** [review]
I've tested several cases, and it works as expected. Great work!
I left some comments.

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/EmailShare/ConfigEmailSharing/EmailListConfiguration.tsx
Comment (juanicabanas) [reply public/app/features/dashboard-scene/sharing/ShareButton/share-externally/EmailShare/ConfigEmailSharing/EmailListConfiguration.tsx:32]: yes, I missed it!

File: public/app/features/dashboard/api/publicDashboardApi.ts
Comment (juanicabanas) [reply public/app/features/dashboard/api/publicDashboardApi.ts:203]: sorry, I missed these messages. However, once we merge this PR we will need help of @imatwawana to review all the texts

Comment (evictorero) [reply public/app/features/dashboard/api/publicDashboardApi.ts:203]: Maybe we want to wait until we have the final text so we don't translate them twice? In that case, we can take care of this in another PR.

Comment (juanicabanas) [reply public/app/features/dashboard/api/publicDashboardApi.ts:203]: There's no "second" translation. I'll do that in this PR so we have the <Trans> component added in these cases. THen, we can change the text and that's all

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/EmailShare/ConfigEmailSharing/ConfigEmailSharing.tsx
Comment (evictorero) [inline public/app/features/dashboard-scene/sharing/ShareButton/share-externally/EmailShare/ConfigEmailSharing/ConfigEmailSharing.tsx:54]: When the public dashboard is paused, everything is disabled except the public/email sharing selection. Is that expected? The previous implementation allowed users to interact with the public dashboard configuration when disabled. 
With this new approach, I think it's unclear why the entire configuration is disabled on a paused public dashboard.

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx
Comment (juanicabanas) [reply public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx:149]: yes, we can review this. I'll add it as part of our tech debt epic. It's out of the scope of this PR

Comment (juanicabanas) [reply public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx:149]: Added [here](https://github.com/grafana/grafana/issues/89118)

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareTypeSelect.tsx
Comment (evictorero) [reply public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareTypeSelect.tsx:102]: This problem is probably related to using a Select component for only two options instead of a radio button. We don't want to block this PR because there are dependencies but we are going to work on this soon.

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/EmailShare/ConfigEmailSharing/ConfigEmailSharing.tsx
Comment (evictorero) [reply public/app/features/dashboard-scene/sharing/ShareButton/share-externally/EmailShare/ConfigEmailSharing/ConfigEmailSharing.tsx]: This was changed, now a paused dashboard configuration is fully available.

File: public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx
Comment (evictorero) [inline public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx:109]: The `NoUpsertPermissionsAlert` warning should be displayed at the top, not in the middle. I couldn't find the design for it.

Comment (juanicabanas) [reply public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx:109]: there was a design, so I did it, but I can change it. I think it makes sense to be at the top. I'll add it to the ShareAlerts component

**Review by evictorero — APPROVED** [review]
LGTM! Tested and work as expected. Great work!

**Pull request merged**
